submissions/Siusarna/data-structures/server/controllers.js

```diff
@@ -0,0 +1,110 @@
+function isJSON(req) {
+  return req.headers['content-type'] === 'application/json';
+}
+
+function showList(res, list) {
+  res.writeHead(200, { 'Content-Type': 'application/json' });
+  res.end(JSON.stringify({ data: list.show() }));
+}
+
+function deleteElementsFromStack(res, stack) {
+  try {
+    const removedValue = stack.pop();
+    res.writeHead(200, { 'Content-Type': 'application/json' });
+    res.end(JSON.stringify({ data: removedValue }));
+  } catch (err) {
+    res.writeHead(400, err.message);
+    res.end();
+  }
+}
+
+function isValidData(inputData) {
+  return typeof inputData === 'number' || typeof inputData === 'string';
+}
+
+function wrongTypeOfValue(res) {
+  res.writeHead(400, 'wrong type of input value');
+  res.end();
+}
+
+function wrongTypeOfContent(res) {
+  res.writeHead(400, 'wrong type of content');
+  res.end();
+}
+
+function insertInList(res, list, inputData) {
```

Comment:
Function called insertInList, but it will also send response, so name of function is confusing right now and function itself violates single responsibility principle.
Please, try to separate this logics. As an example, your function can handle adding and responses must be sent in some other place.